### ISSUE TITLE
Replication Problem with collaborative_filtering_movielens.py script

### ISSUE THREAD

**Issue by santhoshgits** [keras-team/keras-io#833]
Hi,
Is the code 'collaborative_filtering_movielens.py' correctly implemented.
I have replicated this code for same movie datasets. Seems like the prediction accuracy is close to 10%, while the loss is decreasing.

Also, please look at line 203. Does this line pick sorted entry?

**Comment by sachinprasadhs** [comment]
Could you please refer to the latest tutorial migrated to Keras 3 and let us know if you are still facing the error.
https://keras.io/examples/structured_data/collaborative_filtering_movielens/
